### ISSUE TITLE
[MultiSelect.Filterable] Form control with "combobox" role has no associated label

### ISSUE THREAD

**Issue by kinueng** [carbon-design-system/carbon#8567]
## Environment

> Operating system

macOS 11.3

> Browser

Chrome Version 90.0.4430.93 (Official Build) (x86_64)

> Assistive technology used to verify

IBM Equal Access Accessibility Checker v3.1.6

## Detailed description

> What version of the Carbon Design System are you using?

carbon-components
10.34.0
carbon-components-react
7.34.0

> What WCAG 2.1 checkpoint does the issue violate?

https://www.w3.org/TR/WCAG20-TECHS/H44

## Steps to reproduce the issue

1. See source code https://codesandbox.io/s/h4ik6?file=/src/App.js
2. See code rendering from step 1 https://h4ik6.csb.app/
3. Run IBM Equal Access Accessibility Checker on step 2
4. Observe violation Form control with "combobox" role has no associated label
![image](https://user-images.githubusercontent.com/31117513/116921684-eae57c00-ac19-11eb-963c-6a5c42f2bda7.png)

**Comment by tw15egan** [comment]
This error is thrown due to the missing label. The current workaround is to add a label, and if needed, hide it visually but ensure it can still be read by screen readers.

We recently added a `hideTitleText` to `Dropdown`, so we may need to add the same functionality here

ref https://github.com/carbon-design-system/carbon/pull/7594
ref https://github.com/carbon-design-system/carbon/issues/7571

**Comment by lavenal** [comment]
Thanks, I've implemented the suggested workaround with a hidden label using the  bx--visually-hidden class and the violation is gone.

**Comment by kinueng** [comment]
Thank you @tw15egan for the workaround suggestion.
Thank you @lavenal for driving the work on our side.
